ClientApp/src/components/UserApp/User/MyAccount/UpdateEmail.tsx

```diff
@@ -0,0 +1,172 @@
+                setCookie(
+                    'user',
+                    {
+                      token: cookie.user.token,
+                      id: cookie.user.id,
+                      email: newEmail1.value,
+                      name: cookie.user.name,
+                      gender: cookie.user.gender,
+                      role: cookie.user.role
+                    },
+                    { path: '/', maxAge: 31536000 }
+                  );
+            }
+        }
+      };
+
+    const classes = useStyles();
+
+    return (
+        <Grid container spacing={2}>
+            <Grid item xs={12}>
+                <TextField
+                    error={newEmail1.error}
+                    variant="outlined"
+                    fullWidth
+                    value={newEmail1.value}
+                    name="newEmail1"
+                    label="Naujas el. paštas"
+                    id="newEmail1"
+                    helperText={newEmail1.helperText}
+                    onChange={e =>
+                        setNewEmail1({ value: e.target.value, error: false })
+                    }
+                />
+            </Grid>
+            <Grid item xs={12}>
+                <TextField
+                    error={newEmail2.error}
+                    variant="outlined"
+                    fullWidth
+                    value={newEmail2.value}
+                    name="newEmail2"
+                    label="Pakartokite naujo el. pašto adresą"
+                    id="newEmail2"
+                    helperText={newEmail2.helperText}
+                    onChange={e =>
+                        setNewEmail2({ value: e.target.value, error: false })
+                    }
+                />
+            </Grid>
+            <Grid item xs={12}>
+                <Button
+                    fullWidth
+                    type="submit"
+                    variant="contained"
+                    color="primary"
+                    className={classes.submit}
+                    onClick={() => handleSubmit()}
+                >
+                    Patvirtinti
+                </Button>
+            </Grid>
+            <Snackbar
+                anchorOrigin={{
+                    vertical: 'top',
+                    horizontal: 'center'
+                }}
+                open={snackbarText !== ''}
+                autoHideDuration={6000}
+                onClose={() => setSnackbarText('')}
+                message={<span>{snackbarText}</span>}
+            />
+         </Grid>
+        
+    );
+};
+
+const sendForm = async (url: string, obj: IEmailData, token: string): Promise<Response> => {
+    const response: Response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+         Authorization: 'Bearer ' + token
+      },
+      body: JSON.stringify(obj)
+    });
+    return response;
+};
+
+const isEmail = (email: string) => {
```

Comment:
Kadangi šita funckija naudojama jau antrą kartą, tai galėjai perkelti į 'scripts' folderį ir importuoti visur kur naudojama, kad neperrašinėti to pačio kodo.